File: admin-panel/backend-dotnet/src/FuelFlow.Api/Infrastructure/Services/TwilioService.cs
using Twilio;
using Twilio.Rest.Api.V2010.Account;
using Twilio.Types;

namespace FuelFlow.Api.Infrastructure.Services;

public interface ITwilioService
{
    Task<bool> SendVerificationCodeAsync(string phoneNumber, string code);
}

public class TwilioService : ITwilioService
{
    private readonly string? _accountSid;
    private readonly string? _authToken;
    private readonly string? _phoneNumber;
    private readonly bool _isConfigured;

    public TwilioService(IConfiguration configuration)
    {
        _accountSid = configuration["Twilio:AccountSid"] 
            ?? Environment.GetEnvironmentVariable("TWILIO_ACCOUNT_SID");
        
        _authToken = configuration["Twilio:AuthToken"] 
            ?? Environment.GetEnvironmentVariable("TWILIO_AUTH_TOKEN");
        
        _phoneNumber = configuration["Twilio:PhoneNumber"] 
            ?? Environment.GetEnvironmentVariable("TWILIO_PHONE_NUMBER");

        _isConfigured = !string.IsNullOrEmpty(_accountSid) && !string.IsNullOrEmpty(_authToken) && !string.IsNullOrEmpty(_phoneNumber);

        if (_isConfigured)
        {
            TwilioClient.Init(_accountSid, _authToken);
        }
    }

    public async Task<bool> SendVerificationCodeAsync(string phoneNumber, string code)
    {
        if (!_isConfigured)
        {
            Console.WriteLine($"Twilio not configured. Would send code {code} to {phoneNumber}");
            return true;
        }

        try
        {
            var message = await MessageResource.CreateAsync(
                body: $"Your Fuel Flow verification code is: {code}",
                from: new PhoneNumber(_phoneNumber!),
                to: new PhoneNumber(phoneNumber)
            );

            return message.ErrorCode == null;
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Twilio error: {ex.Message}");
            return false;
        }
    }
}

File: admin-panel/backend/src/services/twilio.service.ts
import twilio from 'twilio';

const accountSid = process.env.TWILIO_ACCOUNT_SID;
const authToken = process.env.TWILIO_AUTH_TOKEN;
const twilioPhoneNumber = process.env.TWILIO_PHONE_NUMBER;

if (!accountSid || !authToken || !twilioPhoneNumber) {
    console.warn('⚠️  Twilio credentials not configured. SMS functionality will be disabled.');
}

const client = accountSid && authToken ? twilio(accountSid, authToken) : null;

export class TwilioService {
    /**
     * Send SMS verification code
     */
    async sendVerificationCode(phoneNumber: string, code: string): Promise<boolean> {
        if (!client) {
            console.error('Twilio client not initialized');
            return false;
        }

        try {
            const message = await client.messages.create({
                body: `Your Fuel Flow verification code is: ${code}. Valid for 10 minutes.`,
                from: twilioPhoneNumber,
                to: phoneNumber,
            });

            console.log(`✅ SMS sent successfully to ${phoneNumber}. SID: ${message.sid}`);
            return true;
        } catch (error: any) {
            console.error('❌ Twilio SMS error:', error.message);

            // Handle common errors
            if (error.code === 21211) {
                console.error('Invalid phone number format. Use E.164 format (e.g., +380XXXXXXXXX)');
            } else if (error.code === 21608) {
                console.error('Phone number is not verified. Add it to verified numbers in Twilio console.');
            }

            return false;
        }
    }

    /**
     * Generate a 6-digit verification code
     */
    generateVerificationCode(): string {
        return Math.floor(100000 + Math.random() * 900000).toString();
    }
}

export const twilioService = new TwilioService();
